### PR TITLE
Add AWS_PROFILE needed by some tools

### PR DESCRIPTION
**Pull request by include**
Hi, this little change add profile to the storage which maybe useful to others too.

### CODE REVIEW

File: cmdAdd.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ func getCmdAdd() *cobra.Command {
 		Run:   actionCmdAdd,
 	}
 
+	cmd.Flags().StringVarP(&cfg.Add.Profile, "x-profile", "x", "", "the profile name")
```
Comment (Luzifer) [inline cmdAdd.go:18]: Why is this option called `x-profile`? (And why is the shorthand `x`?)

File: credentials/credentialStore.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ type AWSCredentialStore struct {
 
 // AWSCredential holds the credential set for an environment
 type AWSCredential struct {
+    AWSProfile         string
```
Comment (Luzifer) [inline credentials/credentialStore.go:21]: Please use `gofmt` on the source files.

File: shellsupport/bash/bash.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ func bashShellHandler(c credentials.AWSCredential, export bool) []string {
 	}
 
 	return []string{
+		fmt.Sprintf("%s AWS_PROFILE=%s;", flags, c.AWSProfile),
```
Comment (Luzifer) [inline shellsupport/bash/bash.go:22]: As this is not a required parameter for the vast majority of applications the export / variable statement should not be set if the value is empty.

File: cmdAdd.go [hunk 4]
```diff
@@ -30,6 +31,13 @@ func actionCmdAdd(cmd *cobra.Command, args []string) {
 	}
 
 	var err error
+	if len(cfg.Add.Profile) == 0 {
+		cfg.Add.Profile, err = readStdinLine("AWS Profile: ")
```
Comment (Luzifer) [inline cmdAdd.go:35]: This should be marked optional as it is not required for the vast majority of applications.

**Review by Luzifer — CHANGES_REQUESTED** [review]
Aside of the code comments in this review please update all shell exporters, not only one when introducing new environment variables.

Additionally I'm not sure about this. The profiles are indented to be used with a `~/.aws/credentials` file which would encourage the user to put credentials into a plain text file on their computer. This is a behaviour this tool tries to discourage.

Also I've not yet seen any tool requiring the AWS profile feature to be used. Just of curiosity: can you tell me which tool you're implementing this for?

File: cmdAdd.go (hunk 1 above)
Comment (include) [reply cmdAdd.go:18]: dam. missed this...

File: credentials/credentialStore.go (hunk 2 above)
Comment (include) [reply credentials/credentialStore.go:21]: I've used Atom them edited this file on vim :). I am going to fix it.

File: shellsupport/bash/bash.go (hunk 3 above)
Comment (include) [reply shellsupport/bash/bash.go:22]: I can try to do this but... I'm not a go developer. Just wanted to share a little hack.

**Comment by include** [comment]
I've +10 AWS production accounts and I use lots of profiles. They are under .aws/credentials and I want to move them out of there. One of the tools I use is https://github.com/wallix/awless and profiles are used.

**Comment by Luzifer** [comment]
`awsless` does use the variables already set by `awsenv`:
> You basically need your `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` exported in your environment.

Profiles are used if nothing is set to the environment variables…

**Comment by include** [comment]
You are totally right. I can skip it even if some apps rely on it when there are no env vars defined. I am going to use my version with AWS_PROFILE and I'm adding something more useful to it like  AWS_ACCOUNT_ID. Note: I don't need those values secured... I need them managed by some tool and populated in my shell env. But I understand your point. Thanks for issuing that. Oh and by the way, thanks for your tool 👍

**Pull request closed without merging**